### PR TITLE
Adds a reloader task to the default set of Scalatra tasks.

### PR DESCRIPTION
**Pull request by futurechimp**
Nothing fancy, it just replaces the tortuous *~; copy-resources; aux-compile* with *~reloader*.

### CODE REVIEW

**Comment by rossabaker** [comment]
That's a good idea. 

Would we be wise to look at sbt-revolver? It's not quite the same, but scratches a similar itch.

**Comment by futurechimp** [comment]
Hi Ross, yeah I'm looking at sbt-revolver now. I figured I'd get one thing done then try the other :).

**Comment by futurechimp** [comment]
Ok, so we could go with this `~reloader`, or go with sbt-revolver and `~reStart`. The sbt-revolver way is quite a bit more intrusive:

* add the sbt-assembly plugin to either the g8 template or to scalatra-sbt
* add the sbt-revolver plugin to either the g8 template or scalatra-sbt
* add a JettyLauncher object to the g8 template
* add the sbt-revolver settings into the build file
* add the sbt-revolver main class settings into the build file, pointed at the JettyLauncher#main method

Basically, we'd be including the JettyLauncher by default and pointing sbt-revolver and friends at it. I guess it has the advantage of providing a pre-built system in case people want to use sbt-assembly for production, the `reStart` command is much nice than `~;copy-resources;aux-compile`, and it should be a little faster. It doesn't preclude the use of `container:start`, although we'd probably want to at least change the docs to deprecate the use of that (?).

**Comment by casualjim** [comment]
can't you just do ~container:restart with the latest versions of the web plugin?

**Comment by futurechimp** [comment]
@casualjim that works, yeah - I hadn't seen the inclusion of that command. It also seems to reload quicker than the old version used to, are you guys seeing that too?

We can ditch this pull request and go with ~container:restart, and I'll just do a blog post on the use of sbt-revolver.
